G/Graphviz/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Graphviz"
+version = v"2.40.1"
+
+# Collection of sources required to complete build
+sources = [
+    "https://graphviz.gitlab.io/pub/graphviz/stable/SOURCES/graphviz.tar.gz" =>
+    "ca5218fade0204d59947126c38439f432853543b0818d9d728c589dfe7f3a421",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/graphviz-*/
+
+if [[ "${target}" == *-linux-* ]]; then
+    # For some reason the build wasn't able to pick up that the system provided a
```

Comment:
For which platforms did you have this problem?  All Linux platforms I checked (including some of those failing), `./configure` output has this line:
```
checking for sys/stat.h... yes
```
Is the library not using this information (which would be very interesting)?

Reply:
Haha yeah i saw that too. yes, exactly, it seems like the library is not using that information... >.<

I couldn't find exactly what environment variable that `./configure` check was setting, but it doesn't seem to have been picked up. I'm not sure which linux platform I was running, it was whichever the _first one_ the Wizard runs is. I _think_ that's x86-linux-gnu?

This is the line in question:
https://gitlab.com/graphviz/graphviz/blob/75f84028/lib/sfio/sfsetbuf.c#L39
I noticed that on `master`, they're checking a different prepocessor definition symbol, so maybe this has been fixed?
https://gitlab.com/graphviz/graphviz/blob/master/lib/sfio/sfsetbuf.c#L44

I also just realized that i think the version i'm using, 2.40.1, is like 3 years old? That's what's linked from here:
https://graphviz.gitlab.io/_pages/Download/Download_source.html

But on gitlab the latest "stable release" is 2.42.2?
https://gitlab.com/graphviz/graphviz/-/tags/stable_release_2.42.2

Maybe we should update to that?

Reply:
But yeah exactly, i was like -_- how is it still not picking that up if it's right there in the config?

Reply:
Oh, huh, no, yeah, i guess there hasn't been a "release" in 3 years:
https://gitlab.com/graphviz/graphviz/-/releases

Reply:
It looks like that in v2.42.3 they're properly using the `HAVE_SYS_STAT_H` macro provided by autotools, I'm going to update the builder.  It works locally at least for `x86_64-linux-gnu`

Reply:
Cool, yeah, that's what it looked like to me too. Glad it's working without that manual hack now then. Hopefully that also fixes any other weird sys hacks they were doing! :) (which it seems to since macOS builds now).

Thanks!